pkg/networkutils/network.go

```diff
@@ -549,3 +629,134 @@ func isNetworkUnreachable(err error) bool {
 	}
 	return false
 }
+
+// GetRuleList returns IP rules
+func (n *linuxNetwork) GetRuleList() ([]netlink.Rule, error) {
+	return n.netLink.RuleList(unix.AF_INET)
+}
+
+// GetRuleListBySrc returns IP rules whose has matching source IP
+func (n *linuxNetwork) GetRuleListBySrc(ruleList []netlink.Rule, src net.IPNet) ([]netlink.Rule, error) {
+	var srcRuleList []netlink.Rule
+
+	for _, rule := range ruleList {
+		if rule.Src != nil && rule.Src.IP.Equal(src.IP) {
+			srcRuleList = append(srcRuleList, rule)
+		}
+	}
+
+	return srcRuleList, nil
+}
+
+// DelRuleListBySrc deletes IP rules who has matcing source IP
+func (n *linuxNetwork) DelRuleListBySrc(src net.IPNet) error {
```

Comment:
Why not use the full name, `DeleteRuleListBySrc`?

Reply:
ok